server/routes/graph_routes.py

```
@@ -128,6 +128,39 @@ def delete_dependency(user_id, dep_id):
+def get_infrastructure_context_api(user_id):
+    """GET /api/graph/infrastructure/context - Returns the consolidated infrastructure context."""
+    from utils.auth.stateless_auth import get_org_id_for_user
+    from utils.db.connection_pool import db_pool
+
+    org_id = get_org_id_for_user(user_id)
+    if not org_id:
+        return jsonify({"error": "No organization context"}), 400
+
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.13)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)